Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -237,173 +156,92 @@ CFStringRef _CFStringFromLocalizedString(IDWriteLocalizedStrings* localizedStrin
+ *
+ * Note: This function currently uses a cache, meaning that fonts installed during runtime will not be reflected,
+ * unless registered to the user font collection
  */
-HRESULT _DWriteCreateTextFormat(const wchar_t* fontFamilyName,
-                                DWRITE_FONT_WEIGHT weight,
-                                DWRITE_FONT_STYLE style,
-                                DWRITE_FONT_STRETCH stretch,
-                                float fontSize,
-                                IDWriteTextFormat** outTextFormat) {
-    // Get the direct write factory instance
-    ComPtr<IDWriteFactory> dwriteFactory;
-    RETURN_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
-
-    return dwriteFactory->CreateTextFormat(fontFamilyName,
-                                           nullptr,
-                                           weight,
-                                           style,
-                                           stretch,
-                                           fontSize,
-                                           __GetUserDefaultLocaleName().data(),
-                                           outTextFormat); //
+CFStringRef _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
+    return _DWriteGetFontPropertiesFromName(fontName)->familyName.get();
 }
 
 /**
  * Helper that creates a IDWriteFontFamily object for a given family name
  */
-HRESULT _DWriteCreateFontFamilyWithName(CFStringRef familyName, IDWriteFontFamily** outFontFamily) {
-    ComPtr<IDWriteFactory> dwriteFactory;
-    RETURN_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
+HRESULT _DWriteCreateFontFamilyWithName(CFStringRef familyName, _Out_ IDWriteFontFamily** outFontFamily) {
+    const auto& familyNameUnichars = Strings::VectorFromCFString(familyName);
 
-    ComPtr<IDWriteFontCollection> systemFontCollection;
-    RETURN_IF_FAILED(dwriteFactory->GetSystemFontCollection(&systemFontCollection));
-
-    CFIndex familyNameLength = CFStringGetLength(familyName);
-    std::vector<UniChar> unicharFamilyName(familyNameLength + 1);
-    CFStringGetCharacters(familyName, CFRangeMake(0, familyNameLength), unicharFamilyName.data());
-
-    size_t fontFamilyIndex;
-    BOOL fontFamilyExists;
-
-    RETURN_IF_FAILED(systemFontCollection->FindFamilyName(reinterpret_cast<const wchar_t*>(unicharFamilyName.data()),
-                                                          &fontFamilyIndex,
-                                                          &fontFamilyExists));
-
-    if (fontFamilyExists) {
-        return systemFontCollection->GetFontFamily(fontFamilyIndex, outFontFamily);
+    HRESULT result =
+        __GetSystemFontCollectionHelper()->CreateFontFamilyWithName(reinterpret_cast<const wchar_t*>(familyNameUnichars.data()),
+                                                                    outFontFamily);
+    if (result != S_FALSE) {
```

Comment:
no, this is also intended to be an early return if the call failed above.

---
In reply to: [90911452](https://github.com/Microsoft/WinObjC/pull/1504#discussion_r90911452) [](ancestors = 90911452)